src/main.rs

```diff
@@ -56,6 +57,7 @@ async fn main() -> std::io::Result<()> {
             // Enable Governor middleware
             .wrap(Governor::new(&governor_conf))
             .app_data(web::Data::new(pool.clone()))
+            .service(status_get_api)
```

Comment:
This would be replaced by:

```rust
.service(web::scope("/api").configure(api_scoped_config))
```

Rationale: We want to scope all `/api` paths to an api config.

Reply:
@ChellappanRajan 

Also in `main.rs`, we can create a scoped config for `/api` routes, just below the already existing `scoped_config`. The final API url would be `localhost:8082/api/v1/stats`.

```rust
fn scoped_config(cfg: &mut web::ServiceConfig) {
    cfg.service(index_page);
}

fn api_scoped_config(cfg: &mut web::ServiceConfig) {
    cfg.service(status_get_api);
}
```

Reply:
Fixed. Can you please explain the advantage of using this way `service(web::scope("/api").configure(api_scoped_config))`

Reply:
One small advantage is everything with the path prefix `/api` would be handled by a particular config, rather than we adding all the handlers for `/api` `HttpServer::new` object creation. `main` function would be a little more clean. Though there can be different and better ways to handle this as well.